protocols/directfund/directfund.go

```diff
@@ -203,12 +204,24 @@ func (o Objective) Update(event protocols.ObjectiveEvent) (protocols.Objective,
 
 	updated := o.clone()
 	updated.C.AddSignedStates(event.SignedStates)
-	if event.Holdings != nil && event.BlockNum > updated.latestBlockNumber {
-		updated.C.OnChainFunding = event.Holdings.Clone()
-		updated.latestBlockNumber = event.BlockNum
+
+	return &updated, nil
+}
+
+func (o Objective) UpdateWithChainEvent(event chainservice.Event) (protocols.Objective, error) {
+	updated := o.clone()
+
+	de, ok := event.(chainservice.DepositedEvent)
+	if !ok {
+		return &updated, fmt.Errorf("objective %+v cannot handle event %+v", updated, event)
+	}
+	if de.Holdings != nil && de.GetBlockNum() > updated.latestBlockNumber {
```

Comment:
how reorg safe is this check?

I'm not well informed on the specifics, but is a check against total difficulty a better approach than blocknum?

Do we intend to rely on the chain service to apply these checks, or should it be in-protocol?

Reply:
These are great questions! We should reevaluate our chain reorg strategy. That work is outside the scope of this PR, so I will file a new issue.

Reply:
Filed https://github.com/statechannels/go-nitro/issues/549